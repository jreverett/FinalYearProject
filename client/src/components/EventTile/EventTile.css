.card {
  cursor: pointer;
}

.start-time-label {
  margin-bottom: 0.25rem;
  color: rgb(255, 79, 79);
  font-weight: bold;
}

.cost-label {
  margin-bottom: 0.25rem;
  color: rgb(190, 177, 54);
  font-weight: bold;
}

.cost-label.free-event {
  color: rgb(22, 163, 29);
}

.card-image {
  width: 100%;
  max-height: 150px;
  object-fit: cover;
}
