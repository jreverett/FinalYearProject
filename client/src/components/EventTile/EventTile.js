import React, { Component } from 'react';
import { Card } from 'react-bootstrap';
import moment from 'moment';

import './EventTile.css';

class EventTile extends Component {
  render() {
    let event = this.props.eventDetails;

    let startDate = formatDateTime(event.start);

    // check if an image for this event exists, otherwise use the placeholder
    var thumbnail = event.images[0]
      ? 'data:image/png;base64,' + event.images[0]
      : require('../../images/event-thumb-placeholder.png');

    return (
      <Card border="primary">
        <Card.Img className="card-image" variant="top" src={thumbnail} />
        <Card.Body>
          <p className="start-time-label">{startDate}</p>
          <p className={'cost-label ' + (!event.cost ? 'free-event' : null)}>
            {event.cost ? 'Â£' + event.cost : 'FREE!'}
          </p>
          <Card.Title>{event.title}</Card.Title>
          <Card.Text>{event.address.description}</Card.Text>
        </Card.Body>
      </Card>
    );
  }
}

function formatDateTime(time) {
  // convert ISO8601 datetime to string e.g "Fri, Jun 1, 18:00" update this
  return moment(time).format('h:mm a, ddd Do MMM, YYYY');
}

export default EventTile;
