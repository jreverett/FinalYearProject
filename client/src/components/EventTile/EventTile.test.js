import React from 'react';
import renderer from 'react-test-renderer';
import EventTile from './EventTile';

describe('<EventTile />', () => {
  it('matches the snapshot', () => {
    const testEvent = {
      address: { description: 'Test Lane' },
      start: '2030-01-01T12:34:56.000+00:00',
      subscribers: [],
      images: [],
    };
    const tree = renderer
      .create(<EventTile eventDetails={testEvent} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
