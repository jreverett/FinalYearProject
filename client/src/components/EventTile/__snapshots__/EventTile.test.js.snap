// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventTile /> matches the snapshot 1`] = `
<div
  className="event-gallery-masonry-brick card border-primary"
  onClick={[Function]}
>
  <img
    className="card-img-top card-image"
    src="event-thumb-placeholder.png"
  />
  <div
    className="card-body"
  >
    <p
      className="start-time-label"
    >
      12:34 pm, Tue 1st Jan, 2030
    </p>
    <p
      className="cost-label null"
    >
      Â£NaN
    </p>
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    >
      Test Lane
    </p>
  </div>
</div>
`;
